File: C#/Fundamentals/ObjectsAndClasses(EXERCISE)/Ex2Articles/Article.cs
namespace Ex2Articles
{
    public class Article
    {
        public string Title { get; set; }
        public string Content { get; set; }
        public string Author { get; set; }

        public Article(string title, string content, string author)
        {
            this.Title = title;
            this.Content = content;
            this.Author = author;
        }

        public void Edit(string newContent)
        {
            Content = newContent;
        }

        public void ChangeAuthor(string newAuthor)
        {
            Author = newAuthor;
        }

        public void Rename(string newTitle)
        {
            Title = newTitle;
        }

        public override string ToString()
        {
            return $"{Title} - {Content}: {Author}";
        }
    }
}

File: JavaScript/Applications/L6-ClientSideRendering/lit-demo/templates/article.js
import { html } from '../node_modules/lit-html/lit-html.js'
import commentTemplate from './comment.js'

const articleTemplate = (onSubmit, data) => html`
<article>
    <input type="text" ?disabled=${data.disabled} .value=${data.colour}>
    <h3>${data.title}</h3>
    <div class=${data.colour}>
        <p>${data.content}</p>
    </div>
    <footer>Author: ${data.author}${data.isOwner ? html`<button>Edit</button>` : null}</footer>
    
    <div class="comments">
        <form @submit=${onSubmit}>
            <textarea name="comment"></textarea>
            <button>Submit comment</button>
        </form>

        <ul>
            ${data.comments.map(commentTemplate)}
        </ul>
    </div>
</article>`;

export default articleTemplate;
